bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/IDFUtil.java

```diff
@@ -75,6 +76,16 @@ public static File getIDFPythonScriptFile()
 				+ IDFConstants.IDF_PYTHON_SCRIPT;
 		return new File(idf_py_script);
 	}
+	
+	/**
+	 * @return idf.py file path based on the IDF_PATH given in the argument
+	 */
+	public static File getIDFPythonScriptFile(String idf_path)
+	{
+		String idf_py_script = idf_path + IPath.SEPARATOR + IDFConstants.TOOLS_FOLDER + IPath.SEPARATOR
+				+ IDFConstants.IDF_PYTHON_SCRIPT;
+		return new File(idf_py_script);
+	}
```

Comment:
Could you call this arg method from the existing implementation to avoid duplicate code. For example,


```
public static File getIDFPythonScriptFile()
	{
		String idf_path = getIDFPath();
		return getIDFPythonScriptFile(idf_path);
	}

```